### PR TITLE
Stomp planner

### PR DESCRIPTION
**Pull request by bbrito**
Same as requested in cob_robots

merge together with https://github.com/ipa320/cob_robots/pull/631

### CODE REVIEW

**Review by fmessmer — CHANGES_REQUESTED** [review]
Please review your PR for formal correctness before we can review the actual feature:
 - do not change indentation scheme
 - revert moving files
 - do not add unintended files
 - reference/link related PRs

File: cob_kinematics/ikfast/plugins.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<library path="ros/lib/libikfast_ur10">
+<class name="cob_kinematics/IKFast_ur10" type="ur10::IKFastPlugin" base_class_type="kinematics::KinematicsBase"/>
+</library>
+<library path="ros/lib/libikfast_lbr">
+<class name="cob_kinematics/IKFast_lbr" type="lbr::IKFastPlugin" base_class_type="kinematics::KinematicsBase"/>
+</library>
+<library path="ros/lib/libikfast_ur5">
+<class name="cob_kinematics/IKFast_ur5" type="ur5::IKFastPlugin" base_class_type="kinematics::KinematicsBase"/>
+</library>
+</class_libraries>
```
Comment (fmessmer) [inline cob_kinematics/ikfast/plugins.xml]: you need to be cautious when using `git add -a` or `git commit -a`...I do not recommend it....
remove this file...

File: cob_moveit_bringup/launch/move_group.launch [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <launch>
   <arg name="robot" default="$(optenv ROBOT !!NO_ROBOT_SET!!)"/>
   <arg name="config_path" default="$(find cob_moveit_config)/$(arg robot)/config"/>
-  <arg name="load_robot_description" default="false"/>
+  <arg name="load_robot_description" default="true"/>
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/move_group.launch]: do not change default behavior!

File: cob_moveit_bringup/launch/move_group.launch [hunk 3]
```diff
@@ -38,6 +39,7 @@
   <include ns="move_group" file="$(find cob_moveit_bringup)/launch/controller_manager.xml" if="$(arg allow_trajectory_execution)">
     <arg name="config_path" value="$(arg config_path)"/>
     <arg name="fake_execution" value="$(arg fake_execution)"/>
+    <arg name="moveit_controller_manager" value="fake" if="$(arg fake_execution)"/>
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/move_group.launch]: `if` condition goes first...
also, fake_execution is only used in demo.launch...please remove

File: cob_moveit_bringup/launch/move_group.launch [hunk 4]
```diff
@@ -54,7 +56,8 @@
     <param name="allow_trajectory_execution" value="$(arg allow_trajectory_execution)"/>
     <param name="max_safe_path_cost" value="$(arg max_safe_path_cost)"/>
     <param name="jiggle_fraction" value="$(arg jiggle_fraction)" />
-
+	<param name="collision_detector" value="IndustrialFCL" />
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/move_group.launch]: indentation

Comment (fmessmer) [reply cob_moveit_bringup/launch/move_group.launch]: I guess people wanting to use this need to install some things....please add the package for `IndustrialFCL` to the package.xml

File: cob_moveit_bringup/launch/moveit_ompl.rviz [hunk 5]
```diff
@@ -5,9 +5,12 @@ Panels:
     Property Tree Widget:
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/moveit_ompl.rviz]: do not commit changes to the rviz config files...keep your configuration local and uncommited
same below

File: cob_moveit_bringup/launch/rviz.launch [hunk 6]
```diff
@@ -2,14 +2,15 @@
...
+  <arg name="pipeline" default="ompl"/>
+  
   <arg name="debug" default="false" />
   <arg unless="$(arg debug)" name="launch_prefix" value="" />
   <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />
 
   <arg name="config" default="false" />
   <arg unless="$(arg config)" name="command_args" value="" />
-  <arg     if="$(arg config)" name="command_args" value="-d $(find cob_moveit_bringup)/launch/moveit.rviz" />
+  <arg     if="$(arg config)" name="command_args" value="-d $(find cob_moveit_bringup)/launch/moveit_$(arg pipeline).rviz" />
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/rviz.launch]: we could make the config arg to accept a filename...

File: cob_moveit_bringup/launch/stomp_planning_pipeline.xml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+<launch>
+  <arg name="robot" default="$(optenv ROBOT !!NO_ROBOT_SET!!)"/>
+  <arg name="config_path" default="$(find cob_moveit_config)/$(arg robot)/config"/>
+  <!-- Stomp Plugin for MoveIt! -->
+  <arg name="planning_plugin" value="stomp_moveit/StompPlannerManager" />
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/stomp_planning_pipeline.xml]: add `stomp_moveit` as a dependency to the package.xml

**Review by fmessmer — CHANGES_REQUESTED** [review]

File: cob_moveit_bringup/package.xml [hunk 8]
```diff
@@ -22,5 +22,7 @@
   <exec_depend>moveit_setup_assistant</exec_depend>
   <exec_depend>robot_state_publisher</exec_depend>
   <exec_depend>tf</exec_depend>
+  <exec_depend>IndustrialFCL</exec_depend>
+  <exec_depend>stomp_moveit</exec_depend>
```
Comment (fmessmer) [inline cob_moveit_bringup/package.xml]: also add `moveit_ros_perception` as `occupancy_map_monitor` as it is used in cob_robots/cob_moveit_config sensors.yaml!

File: cob_moveit_bringup/package.xml [hunk 9]
```diff
@@ -22,5 +22,7 @@
   <exec_depend>moveit_setup_assistant</exec_depend>
   <exec_depend>robot_state_publisher</exec_depend>
   <exec_depend>tf</exec_depend>
+  <exec_depend>IndustrialFCL</exec_depend>
```
Comment (fmessmer) [inline cob_moveit_bringup/package.xml]: IndustrialFCL is part of `industrial_collision_detection` which is not released...
what do you use it for?

File: cob_moveit_bringup/package.xml (hunk 8 above)
Comment (fmessmer) [reply cob_moveit_bringup/package.xml]: also, please respect alphabetical order in package.xml

Comment (fmessmer) [reply cob_moveit_bringup/package.xml]: unfortunately, `stomp_moveit` seems not to be released as well...

File: cob_moveit_bringup/launch/move_group.launch [hunk 10]
```diff
@@ -36,8 +37,9 @@
 
   <!-- Trajectory Execution Functionality -->
   <include ns="move_group" file="$(find cob_moveit_bringup)/launch/controller_manager.xml" if="$(arg allow_trajectory_execution)">
-    <arg name="config_path" value="$(arg config_path)"/>
+    <arg name="moveit_controller_manager" value="fake" if="$(arg fake_execution)"/>
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/move_group.launch]: why do you need this?
isn't the line `<arg name="fake_execution" value="$(arg fake_execution)"/>` sufficient for fake_execution?

File: cob_moveit_bringup/launch/move_group.launch [hunk 11]
```diff
@@ -54,7 +56,8 @@
     <param name="allow_trajectory_execution" value="$(arg allow_trajectory_execution)"/>
     <param name="max_safe_path_cost" value="$(arg max_safe_path_cost)"/>
     <param name="jiggle_fraction" value="$(arg jiggle_fraction)" />
-
+    <param name="collision_detector" value="IndustrialFCL" />
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/move_group.launch:57]: what does this do?

File: cob_moveit_bringup/launch/ompl_rviz.launch [hunk 12]
```diff
@@ -0,0 +1,18 @@
+<launch>
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/ompl_rviz.launch]: remove this file...you need to load the proper config locally

File: cob_moveit_bringup/launch/stomp_moveit.rviz [hunk 13]
```diff
@@ -0,0 +1,329 @@
+Panels:
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/stomp_moveit.rviz]: remove this file...you need to load the proper config locally...it's a duplicate anyway...

File: cob_moveit_bringup/launch/moveit_stomp.rviz [hunk 14]
```diff
@@ -0,0 +1,572 @@
+Panels:
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/moveit_stomp.rviz]: remove this file...you need to load the proper config locally

File: cob_moveit_bringup/launch/moveit_ompl.rviz [hunk 15]
```diff
@@ -0,0 +1,551 @@
+Panels:
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/moveit_ompl.rviz]: remove this file...you need to load the proper config locally

File: cob_moveit_bringup/launch/stomp_moveit_planning_pipeline.launch.xml [hunk 16]
```diff
@@ -0,0 +1,20 @@
+<launch>
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/stomp_moveit_planning_pipeline.launch.xml]: stick to the naming scheme and call this file `stomp_planning_pipeline.xml` in analogy to ompl

File: cob_moveit_bringup/launch/stomp_moveit_rviz.launch [hunk 17]
```diff
@@ -0,0 +1,18 @@
+<launch>
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/stomp_moveit_rviz.launch]: remove this file...you need to load the proper config locally...it's a duplicate anyway

File: cob_moveit_bringup/launch/stomp_moveit_planning_pipeline.launch.xml (hunk 16 above)
Comment (fmessmer) [reply cob_moveit_bringup/launch/stomp_moveit_planning_pipeline.launch.xml]: remove this file...the required `stomp_planning_pipeline.xml` is part of this PR below

**Review by fmessmer — CHANGES_REQUESTED** [review]
please remove the rviz configs as they are robot and scenario specific....keep them local only!

**Comment by fmessmer** [comment]
We have some significant dependency trouble for this feature as they do not seem to be released!

Could you post what additional packages you use in your workspace, i.e.
 - moveit from source? which branch?
 - industrial_moveit from source?
 - stomp? which repo?

File: cob_moveit_bringup/launch/move_group.launch (hunk 11 above)
Comment (bbrito) [reply cob_moveit_bringup/launch/move_group.launch:57]: We need it for stomp....

**Comment by bbrito** [comment]
I used this one:

http://wiki.ros.org/industrial_moveit

https://github.com/ros-industrial/industrial_moveit

Branch: Indigo-devel
I installed it from source.

**Comment by fmessmer** [comment]
@ipa-bfb 
unfortunately, it still does not pass Travis CI tests...

**Comment by bbrito** [comment]
We are close to get it ;)

**Comment by fmessmer** [comment]
maybe it's because `stomp_core` has not [install tags](https://github.com/ros-industrial/industrial_moveit/blob/indigo-devel/stomp_core/CMakeLists.txt#L47-L51)

**Comment by bbrito** [comment]
How should I proceed?

**Comment by mathias-luedtke** [comment]
> How should I proceed?

Add the install and open a PR. Or at least open an issue for pointing it out.

**Comment by fmessmer** [comment]
Well, you could create a fork of `industrial_moveit` and temporarily use the uri of your fork within the .rosinstall file...then apply fixes/changes until Travis CI passes...
Then you create a PR back to ros-industrial with the changes in your fork...asking to be merged and released...

**Comment by bbrito** [comment]
Yes that is a cool idea. I am without computer till Friday but on friday and after the help ipa-mdl i think I can do it tomorrow ;)

**Comment by bbrito** [comment]
Awesome it worked ;) should I make a pull request to them now? Best,
Bruno

PR is https://github.com/ros-industrial/industrial_moveit/pull/32

File: .travis.rosinstall [hunk 18]
```diff
@@ -1,4 +1,4 @@
 - git:
-    uri: 'https://github.com/ipa-bfb/industrial_moveit.git'
+    uri: 'https://github.com/ros-industrial/industrial_moveit.git'
     local-name: industrial_moveit
     version: cmake_install_tags
```
Comment (fmessmer) [inline .travis.rosinstall]: change back the `version` as well

File: cob_moveit_bringup/launch/move_group.launch (hunk 11 above)
Comment (fmessmer) [reply cob_moveit_bringup/launch/move_group.launch:57]: @ipa-bfb 
what package is this plugin(?) provided in?
a `<exec_depend>` needs to be defined for it!

**Review by fmessmer — CHANGES_REQUESTED** [review]

**Comment by bbrito** [comment]
Is provided here:

https://github.com/ros-industrial/industrial_moveit/tree/indigo-devel/industrial_collision_detection

It is part of industrial moveit but it can also be installed directly:

sudo apt-get install ros-indigo-industrial-collision-detection

File: cob_moveit_bringup/launch/move_group.launch (hunk 11 above)
Comment (fmessmer) [reply cob_moveit_bringup/launch/move_group.launch:57]: it's defined in https://github.com/ros-industrial/industrial_moveit/blob/indigo-devel/industrial_collision_detection/industrial_collision_detection_plugin_description.xml
so we need a `<exec_depend>industrial_collision_detection</exec_depend>`

File: cob_moveit_bringup/launch/move_group.launch [hunk 19]
```diff
@@ -15,6 +15,7 @@
   <arg     if="$(arg info)" name="command_args" value="--debug" />
 
   <!-- move_group settings -->
+
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/move_group.launch]: remove blank line

File: cob_moveit_bringup/launch/move_group.launch [hunk 20]
```diff
@@ -36,8 +37,8 @@
 
   <!-- Trajectory Execution Functionality -->
   <include ns="move_group" file="$(find cob_moveit_bringup)/launch/controller_manager.xml" if="$(arg allow_trajectory_execution)">
-    <arg name="config_path" value="$(arg config_path)"/>
     <arg name="fake_execution" value="$(arg fake_execution)"/>
+    <arg name="config_path" value="$(arg config_path)"/>
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/move_group.launch]: restore previous order

File: cob_moveit_bringup/launch/stomp_moveit_planning_pipeline.xml [hunk 21]
```diff
@@ -0,0 +1,20 @@
+<launch>
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/stomp_moveit_planning_pipeline.xml]: remove this file and use `stomp_planning_pipeline.xml` instead

File: cob_moveit_bringup/launch/stomp_planning_pipeline.xml [hunk 22]
```diff
@@ -0,0 +1,23 @@
+<launch>
```
Comment (fmessmer) [inline cob_moveit_bringup/launch/stomp_planning_pipeline.xml:1]: make consistent with `ompl_planning_pipeline.xml`

File: cob_moveit_bringup/package.xml [hunk 23]
```diff
@@ -15,9 +15,11 @@
   <exec_depend>cob_hardware_config</exec_depend>
   <exec_depend>cob_moveit_config</exec_depend>
   <exec_depend>joint_state_publisher</exec_depend>
+  <exec_depend>moveit_ros_perception</exec_depend>
```
Comment (fmessmer) [inline cob_moveit_bringup/package.xml]: remove this - is duplicate

File: cob_moveit_bringup/package.xml [hunk 24]
```diff
@@ -15,9 +15,11 @@
   <exec_depend>cob_hardware_config</exec_depend>
   <exec_depend>cob_moveit_config</exec_depend>
```
Comment (fmessmer) [inline cob_moveit_bringup/package.xml:16]: add `<exec_depend>industrial_collision_detection</exec_depend>`

File: cob_moveit_bringup/package.xml [hunk 25]
```diff
@@ -15,9 +15,11 @@
...
   <exec_depend>joint_state_publisher</exec_depend>
+  <exec_depend>moveit_ros_perception</exec_depend>
   <exec_depend>moveit_planners_ompl</exec_depend>
   <exec_depend>moveit_plugins</exec_depend>
   <exec_depend>moveit_ros_move_group</exec_depend>
+  <exec_depend>moveit_ros_perception</exec_depend>
   <exec_depend>moveit_ros_visualization</exec_depend>
   <exec_depend>moveit_setup_assistant</exec_depend>
   <exec_depend>robot_state_publisher</exec_depend>
```
Comment (fmessmer) [inline cob_moveit_bringup/package.xml:25]: add `<exec_depend>stomp_moveit</exec_depend>`

**Review by fmessmer — CHANGES_REQUESTED** [review]

**Review by fmessmer — APPROVED** [review]

**Pull request merged**